lec15-data-viz.Rmd

```diff
@@ -1,4 +1,341 @@
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length))+
+  geom_point()
+```
+
+You can also save the structure of the graph to a variable and then add layers and other attributes to it. Let's do that:
+```{r}
+g <- ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length)) 
+g + geom_point()
+```
+
+We can add other "layers" of graphs to our first graph. Let's add a trend line to our scatterplot:
+```{r}
+g + geom_point() + 
+  geom_smooth() 
+```
+
+We can also "facet" our graphs, which means dividing a plot into subplots based on the values of one or more discrete variables.
+```{r}
+g + geom_point() + geom_smooth() + facet_grid(.~Species)
+```
+
+There are lots of ways to organize your facet. See "faceting" at the bottom of [ggplot cheetsheet](https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf).
+
+
+Let's say we don't want to facet, but we want to colour the dots depending on the species:
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()
+```
+
+We can change other aesthetic features of the graph. Let's get rid of the color aspect of our structure and change the colour and size of the points:
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length)) + geom_point(color="steelblue", size=4)
+
+```
+
+Going back to our graph coloured by species, what if we want to change other aspects, like the theme, or centre the title?
+
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()+
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  theme_bw() +
+  theme(plot.title = element_text(hjust = 0.5))
+  
+```
+What about changing the font? Let's get rid of the theme for a moment, which changes many aspects of the aesthetics at once. Let's spend some time building our own theme...
+
+```{r}
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()+
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  theme(plot.title = element_text(hjust = 0.5), 
+        text = element_text(size = 8, family='Times'),
+        legend.title=element_blank())
+  
+```
+
+
+You can add your own alterations to themes. For example, here is an APA theme built on top of the base theme_bw(). Themes can be saved into variables so they don't need to be typed out every time you use a graph.
+
+
+```{r}
+
+apatheme <- theme_bw()+
+  theme(panel.grid.major=element_blank(),
+        panel.grid.minor=element_blank(),
+        panel.border=element_blank(),
+        axis.line=element_line(),
+        text=element_text(family='Times'),
+        legend.title=element_blank())
+
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()+
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  apatheme
```

Comment:
Generally it is better to do `apatheme()`

Reply:
good point, changed